### PR TITLE
Allow region-specific UI languages

### PR DESCRIPTION
**Pull request by jiru**
tl;dr: this PR changes the way UI language is encoded in the URL in order to solve #2672.

The existing UI language identifiers have been made to follow ISO-639-3 in order to keep the same language codes as sentences. While this approach sounds right because it makes both UI and corpus rely on the same list, it has limitations:
* we need to define the current UI language in a separate global config variable `Config.language` because it’s different from the actual locale, which is set with `I18n::setLocale()`
* we cannot provide region-specific UI language variants (whereas Transifex allows them, for good reasons), and there was a need for that recently with Taiwanese Chinese in #2672

In order to get rid of these limitations, this PR makes use of [BCP 47](https://en.wikipedia.org/wiki/IETF_language_tag) as UI language identifier. In particular, `cmn` gets changed to `zh-cn`, `por` to `pt-br` and `zh-tw` is added. All previous 3-letter codes are properly redirected, so old URLs work as excepted.

In addition, it looks like this PR solves the (unreported?) issue of font selection for Chinese UI. When Chinese is selected, the tag `<html>` currently has the attribute `lang="zh"` which doesn’t let the browser know if it should choose a font tailored for simplified or traditional characters. With this PR, it now sets it to `lang="zh-cn"` and the font look different if your system is configured with a specific font for "Chinese of mainland China".

### CODE REVIEW

**Comment by jiru** [comment]
Note to self: we also need to change the wiki language identifiers from pt to pt-br and zh to zh-cn, and ideally have a redirection for these wikis.
